executor/adapter.go

```diff
@@ -1805,8 +1806,21 @@ func (a *ExecStmt) SummaryStmt(succ bool) {
 		userString = sessVars.User.Username
 	}
 
+	var enabled, enabledInternalQuery bool
+	if sessVars.EnablePersistentStmtSummary {
+		if sessVars.StmtSummary == nil && stmtsummaryv2.GlobalStmtSummary != nil {
+			sessVars.StmtSummary = stmtsummaryv2.GlobalStmtSummary
+		}
```

Comment:
Will it run into this branch?

Reply:
It is only for some internal sessions, and during testing I found that some internal sessions are created before the `GlobalStmtSummary` is initialized. So here is an opportunity to make them associated to the `GlobalStmtSummary`.